crates/shared/src/price_estimation/competition.rs

```diff
@@ -539,4 +550,77 @@ mod tests {
         let result = racing.estimate(query).await;
         assert_eq!(result.as_ref().unwrap(), &estimate(1));
     }
+
+    #[tokio::test]
+    async fn queries_stages_sequentially() {
+        let query = Arc::new(Query {
+            verification: None,
+            sell_token: H160::from_low_u64_le(0),
+            buy_token: H160::from_low_u64_le(1),
+            in_amount: NonZeroU256::try_from(1).unwrap(),
+            kind: OrderKind::Sell,
+        });
+
+        fn estimate(amount: u64) -> Estimate {
+            Estimate {
+                out_amount: amount.into(),
+                ..Default::default()
+            }
+        }
+
+        let mut first = MockPriceEstimating::new();
+        first.expect_estimate().times(1).returning(move |_| {
+            async {
+                sleep(Duration::from_millis(20)).await;
```

Comment:
Both estimators in the first stage wait for the same period of time before returning their result.
Dropping the `sleep()` from them entirely shouldn't change the result of the test. Was this an oversight or was there some other reason for the `sleeps`?

Reply:
The sleep here is longer than the 10ms sleep below (in the second phase) to make sure we are actually waiting on this stage to complete before polling the next one (otherwise the test would also past without configuring stages)